sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -200,6 +200,38 @@ object SQLConf {
     Map(configs.map { cfg => cfg.key -> cfg } : _*)
   }
 
+  /**
+   * Holds information about keys that have been deprecated.
+   *
+   * @param key The deprecated key.
+   * @param version Version of Spark where key was deprecated.
+   * @param comment Additional info regarding to the removed config. For example,
+   *                reasons of config deprecation, what users should use instead of it.
+   */
+  case class DeprecatedConfig(key: String, version: String, comment: String)
+
+  /**
+   * Maps deprecated SQL config keys to information about the deprecation.
+   *
+   * The extra information is logged as a warning when the SQL config is present
+   * in the user's configuration.
+   */
+  val deprecatedSQLConfigs: Map[String, DeprecatedConfig] = {
+    val configs = Seq(
+      DeprecatedConfig("spark.sql.hive.verifyPartitionPath", "3.0.0",
```

Comment:
@MaxGekk, why don't we use `HIVE_VERIFY_PARTITION_PATH.key`? If forward referencing is the problem, we can simply place those codes on the bottom.

Reply:
and there seem many deprecated configuration more. For example, `spark.sql.execution.arrow.enabled`. do you plan to add it separately?

Reply:
> and there seem many deprecated configuration more.

Right, I added a few of them as an example. I could add all deprecated but this need some time to figure out when a config was deprecated and a reason for deprecation.

Reply:
I roughly checked. Looks there are only 4 instances more (by keyword search "deprecated"). `spark.sql.legacy.execution.pandas.groupedMap.assignColumnsByName` has to be deprecated too FWIW.

Reply:
I added others